app/services/testdata/Phase2TestsInvitedStatusGenerator.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.testdata
+
+import java.util.UUID
+
+import config.CubiksGatewayConfig
+import connectors.testdata.ExchangeObjects.{ CubiksTestResponse, TestGroupResponse }
+import model.persisted.{ CubiksTest, Phase2TestGroup }
+import org.joda.time.DateTime
+import repositories._
+import repositories.onlinetesting.Phase2TestRepository
+import uk.gov.hmrc.play.http.HeaderCarrier
+import config.MicroserviceAppConfig.cubiksGatewayConfig
+import play.api.mvc.RequestHeader
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object Phase2TestsInvitedStatusGenerator extends Phase2TestsInvitedStatusGenerator {
+  override val previousStatusGenerator = Phase1TestsResultsReceivedStatusGenerator
+  override val otRepository = phase2TestRepository
+  override val gatewayConfig = cubiksGatewayConfig
+}
+
+trait Phase2TestsInvitedStatusGenerator extends ConstructiveGenerator {
+  val otRepository: Phase2TestRepository
+  val gatewayConfig: CubiksGatewayConfig
+
+  def generate(generationId: Int, generatorConfig: GeneratorConfig)(implicit hc: HeaderCarrier, rh: RequestHeader) = {
+
+    val etray = CubiksTest(
+      cubiksUserId = scala.util.Random.nextInt(1000000000),
```

Comment:
This int will occasionally overflow. Suggest using Int.MaxValue. In theory, it should be Int.MaxValue + 1, but that just looks scruffy